answer_rocket/data.py

```diff
@@ -878,7 +968,7 @@ def run_max_sql_gen(self, dataset_id: UUID, pre_query_object: Dict[str, any], co
 
             if result.success:
                 result.sql = run_max_sql_gen_response.sql
-                result.df = create_df_from_data(run_max_sql_gen_response.data)
+                result.df = create_df_from_data(run_max_sql_gen_response.data) if execute_sql else None
```

Comment:
@gbarchard This is a code change (not a docstring change), so just pointing it out explicitly.